packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -129,78 +279,94 @@ async function main() {
       const n = new NetworkModel();
       n.name = "nw" + generateString(5);
       n.ip_range = "10.238.0.0/16";
-      n.addAccess = true;
+      n.addAccess = false;
 
-      // create VMs Object for each vm
       const vms = new MachinesModel();
       vms.name = "batch" + (batch + 1);
       vms.network = n;
       vms.machines = [vm];
       vms.metadata = "";
-      vms.description = "Test deploying vm with name " + vm.name + " via ts grid3 client - Batch " + (batch + 1);
+      vms.description = `Test deploying vm with name ${vm.name} - Batch ${batch + 1}`;
 
       batchVMs.push(vms);
     }
 
     const allTwinDeployments: TwinDeployment[] = [];
 
-    const deploymentPromises = batchVMs.map(async (vms, index) => {
+    const deploymentPromises = batchVMs.map(async vms => {
       try {
-        const [twinDeployments, _, __] = await grid3.machines._createDeployment(vms);
-        return { twinDeployments, batchIndex: index };
+        const [twinDeployments] = await grid3.machines._createDeployment(vms);
+        return twinDeployments;
       } catch (error) {
         log(`Error creating deployment for batch ${batch + 1}: ${error}`);
-        return { twinDeployments: null, batchIndex: index };
+        return [];
       }
     });
     console.time("Preparing Batch " + (batch + 1));
     const deploymentResults = await Promise.allSettled(deploymentPromises).then(results =>
       results.flatMap(r => (r.status === "fulfilled" ? r.value : [])),
     );
-    console.timeEnd("Preparing Batch " + (batch + 1));
-
-    for (const { twinDeployments } of deploymentResults) {
-      if (twinDeployments) {
-        allTwinDeployments.push(...twinDeployments);
-      }
-    }
-
-    try {
-      await grid3.machines.twinDeploymentHandler.handle(allTwinDeployments);
-      log(`Successfully handled and saved contracts for some twin deployments`);
-    } catch (error) {
-      errors.push(error);
-      failedCount += batchSize;
-      log(`Error handling contracts for twin deployments: ${error}`);
+    allTwinDeployments.push(...deploymentResults);
+    let batchSuccessfulNodes: Set<number> = new Set();
+    let batchFailedNodes: Set<number> = new Set();
+    if (allTwinDeployments.length > 0) {
+      const handlePromises = [handle(grid3, allTwinDeployments)];
+
+      const results = await Promise.allSettled(handlePromises);
+
+      results.forEach(result => {
+        if (result.status === "fulfilled") {
+          const { resultContracts, successfulNodes, failedNodes } = result.value;
+          batchSuccessfulNodes = successfulNodes;
+          batchFailedNodes = failedNodes;
+
+          successfulNodes.forEach(node => allSuccessfulNodes.push(node));
+          failedNodes.forEach(node => allFailedNodes.add(node));
+
+          log(`Successfully handled deployments for Batch ${batch + 1}`);
+        } else {
+          errors.push(`Error handling deployments for Batch ${batch + 1}: ${result.reason}`);
```

Comment:
should we handle the rejected one to? i mean by pusshing it to the failed deployements